lib/https.js

```diff
@@ -25,8 +25,10 @@ require('internal/util').assertCrypto();
 
 const tls = require('tls');
 const url = require('url');
-const http = require('http');
 const util = require('util');
+const agent = require('_http_agent');
+const server = require('_http_server');
```

Comment:
Just a nit but could these be:

```js
const { Agent: HttpAgent } = require('_http_agent');
const {
  Server: HttpServer,
  _connectionListener: httpConnectionListener
} = require('_http_server');
```

It feels a bit clearer to me that way when you can go `Server.prototype.setTimeout = HttpServer.prototype.setTimeout;`